medicines/doc-index-updater/src/document_manager/mod.rs

```diff
@@ -27,20 +33,36 @@ async fn del_document_handler(
         let duration = Duration::days(1);
 
         match queue.send(message, duration).await {
-            Ok(_) => Ok(warp::reply::json(
-                &state_manager.set_status(id, JobStatus::Accepted).await?,
-            )),
+            Ok(_) => Ok(state_manager.set_status(id, JobStatus::Accepted).await?),
             Err(_) => Err(warp::reject::custom(FailedToDispatchToQueue)),
         }
     } else {
         Err(warp::reject::custom(FailedToDispatchToQueue))
     }
 }
 
+async fn del_document_xml_handler(
+    document_content_id: String,
+    state_manager: StateManager,
+) -> Result<Xml, Rejection> {
+    let r: XMLJobStatusResponse = del_document_handler(document_content_id, state_manager)
+        .await?
+        .into();
```

Comment:
@m-doughty I had a go at making this use an idiomatic `into()`  - I think it's neater than calling the custom `from_…` method

Reply:
I like this.